docs/src/charm/howto/openstack.md

```diff
@@ -0,0 +1,104 @@
+# Integrating with OpenStack
+
+This guide explains how to integrate {{product}} with the OpenStack cloud
+platform.
+
+## Prerequisites
+
+To follow this guide, you will need:
+
+- An [OpenStack][openstack] cloud environment.
+- Juju set-up as the deployment tool for Openstack. Please refer to
+  [Juju and Openstack][juju-openstack] documentation for more information.
+- An OpenStack [project][project] with the necessary permissions to deploy
+  {{product}}.
+- A juju [controller][controller] with [access][credentials] to the OpenStack 
+  cloud environment.
+- A juju [model][model] for deploying {{product}} on OpenStack.
+- Configured [proxy configuration][proxy] in constrained environments.
+
+## Deploying {{product}} on OpenStack
+
+
+To deploy the {{product}} [bundle][bundle] on OpenStack you need an overlay
+bundle which serves as an extension of the core bundle. The overlay bundle
+contains the necessary configuration to deploy {{product}} on OpenStack.
+
+Refer to the base overlay [openstack-lb-overlay.yaml][openstack-overlay] and
+modify it as needed.
+
+### OpenStack overlay configurations:
+
+Run `openstack project list` to retrieve your project id and include the
+[project id][project] in the overlay template:
+
+```yaml
+applications:
+  openstack-integrator:
+    options:
+      project-id: <my-openstack-project-id>
+```
+
+
+Adjust [easyrsa][easyrsa] to avoid creating an LXD machine:
```

Comment:
@HomayoonAlimohammadi Is this necessary?

Reply:
I don't think so. Maybe @addyess has a better idea but I think it's not a very special thing. I mean the problem in our envs is that we can not create LXD machines, maybe that's not the case for the user. And what if something else is going wrong with the Openstack deployment of the user? e.g. What if - let's say - Keystone is not configured correctly? Also seems like [Openstack on LXD is supported](https://docs.openstack.org/charm-guide/victoria/openstack-on-lxd.html) (tho it's only for testing/development purposes).

Reply:
And even if _not using LXD_ is the way to go when deploying on Openstack, I think we should change the [overlay](https://github.com/charmed-kubernetes/bundle/blob/main/overlays/openstack-lb-overlay.yaml) rather than instructing the user to the change the overlay manually.

Reply:
Yeah, i don't think this is critical.  I don't know if we ever got to a solution why the openstack instance we're using for testing requires easyrsa to be its own machine.  I think we shouldn't note "avoiding creating this LXD machine"